hugo/content/docs/metrics.md

```diff
@@ -214,37 +50,129 @@ https://github.com/asynkron/protoactor-dotnet/issues/948
+It is also possible to setup OpenTelemetry Protocol exporter depending which backend is used. In order to reuse below example, `OpenTelemetry.Exporter.OpenTelemetryProtocol` nuget package is needed.
+It is suggested also to configure  `PeriodicExportingMetricReaderOptions` if it is needed to control how fast metrics are exported (default value is 60 seconds).
+
+`OpenTelemetry` metrics in C# implementation use [System.Diagnositcs.Metrics](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/metrics-instrumentation).
+`AddProtoActorInstrumentation()` extension shown in the example is adding Proto.Actor meter name.
+Method has additional parameter `useRecommendedHistogramBoundaries` which is true by default. It causes that default buckets used in OpenTelemetry implementation are changed to more preffered ones.
 
-## Running monitoring systems using Docker
+```csharp
+void ConfigureMetrics(WebApplicationBuilder builder) =>
+    builder.Services.AddOpenTelemetryMetrics(b =>
+        b.SetResourceBuilder(ResourceBuilder
+                .CreateDefault()
+                .AddAttributes(new KeyValuePair<string, object>[]
+                {
+                    new("someLabel", builder.Configuration["SomeLabel"]),
+                    new("env", builder.Environment.EnvironmentName)
+                })
+                .AddService(builder.Configuration["Service:Name"])
+            )
+            .AddProtoActorInstrumentation()
+            .AddOtlpExporter(opt =>
+            {
+                opt.Endpoint = new Uri(builder.Configuration["Otlp:Endpoint"]);
+                opt.PeriodicExportingMetricReaderOptions.ExportIntervalMilliseconds =
+                    builder.Configuration.GetValue<int>("Otlp:MetricsIntervalMilliseconds");
+            })
+    );
 
-In this section, we will examine how to run Prometheus and Grafana locally using Docker. This is very easy to do, just run one Docker command.
+```
 
-### Running Prometheus
+## Using Prometheus and Grafana to store and visualize metrics
+
+Another example of Prometheus metrics setup might be found in [ActorMetrics example](https://github.com/asynkron/protoactor-dotnet/tree/dev/examples/ActorMetrics).
+This one doesn't use OpenTelemetry Exporter and the application has own [endpoint for Prometheus metrics](https://github.com/asynkron/protoactor-dotnet/blob/dev/examples/ActorMetrics/Startup.cs#L29).
+
+In this section, it will be presented how to run Prometheus and Grafana locally using Docker. In the setup showned above, it is also required to run [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) to be able to save metrics in Prometheus.
+
+```yml
+version: '3.7'
+
+services:     
+  grafana:
+    image: grafana/grafana
+    ports:
+      - 3000:3000
+      
+  prometheus:
+    image: prom/prometheus
+    volumes:
+      - ./prometheus.yml:/etc/prometheus/prometheus.yml
+    ports:
+      - 9090:9090
+      
+      
+  otel-collector:
```

Comment:
Perhaps we don't need to complicate this example by using otel-collector. I think that it would be easier if in the documentation we just used the in-proc prometheus exporter. Realtime map can be mentioned here as a alternative approach with otel-collector and people who are interested in it will look it up. Realtime map uses otel-collector to easily switch between Prometheus (on local dev) and Prometheus remote write to Grafana Cloud (on prod).

Reply:
If we want to keep documentation as simple as possible then I agree that Prometheus exporter is enough. I can also change the example above to align with the ActorMetrics sample. I will link the realtime map docker compose to show how to setup oltp locally.

https://github.com/asynkron/realtimemap-dotnet/blob/ccaa9099f5a6cae615feabd38c3cfcc08e791a6f/devenv/docker-compose.yml#L36

Reply:
I haven't noticed that realtime map has it already.